src/DesktopIntegration.vala

```diff
@@ -107,12 +107,55 @@ public class Gala.DesktopIntegration : GLib.Object {
         foreach (unowned var app in apps) {
             foreach (weak Meta.Window window in app.get_windows ()) {
                 if (window.get_id () == uid) {
-                    window.get_workspace ().activate_with_focus (window, wm.get_display ().get_current_time ());
+                    if (window.has_focus ()) {
+                        notify_already_focused (window);
+                    } else {
+                        window.get_workspace ().activate_with_focus (window, wm.get_display ().get_current_time ());
+                    }
                 }
             }
         }
     }
 
+    private bool notifying = false;
+    private void notify_already_focused (Meta.Window window) {
+        if (notifying) {
+            return;
+        }
+
+        notifying = true;
+
+        wm.get_display ().get_sound_player ().play_from_theme ("bell", _("Window has already focus"), null);
```

Comment:
Can we not use Gdk.Beep in Gala?

Reply:
Unfortunately not, we get a glib critical along the lines of `assertion Gdk.Display != null failed` :(

Reply:
yeah no that's better to use Mutter there